mailbox/store/src/main/java/org/apache/james/mailbox/store/event/SynchronousEventDelivery.java

```diff
@@ -19,23 +19,37 @@
 
 package org.apache.james.mailbox.store.event;
 
+import javax.inject.Inject;
+
 import org.apache.james.mailbox.Event;
 import org.apache.james.mailbox.MailboxListener;
+import org.apache.james.metrics.api.MetricFactory;
+import org.apache.james.metrics.api.TimeMetric;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class SynchronousEventDelivery implements EventDelivery {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(SynchronousEventDelivery.class);
 
+    private final MetricFactory metricFactory;
+
+    @Inject
+    public SynchronousEventDelivery(MetricFactory metricFactory) {
+        this.metricFactory = metricFactory;
+    }
+
     @Override
     public void deliver(MailboxListener mailboxListener, Event event) {
+        TimeMetric timer = metricFactory.timer("mailbox-listener-" + mailboxListener.getClass().getSimpleName());
```

Comment:
Next time, it will be great to add a single MetricConstants interface in order to have all metric prefixes in a single file.